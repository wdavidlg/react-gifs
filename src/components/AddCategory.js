import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = (props) => {


    const [value, setValue] = useState('');

    const handleChange = (e) => {
        setValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        //console.log('handleSubmit', value);
        if(value.trim().length !== 0){
            props.setCategories((prev) => [value, ...prev]);
            setValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
           <input 
                type="text"
                placeholder="Add Category"
                value={value}
                onChange={handleChange}
           />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
