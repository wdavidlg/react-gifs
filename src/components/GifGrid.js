import React from 'react'
import Gif from './Gif'
import { useFetchGifs } from '../hooks/useFetchGifs'
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
    

    const {data:gifs, loading} = useFetchGifs(category);

    const getCardGifs = () => {
        return gifs.map((item) => <Gif key={item.id} gif={item} />);
    }
    console.log(loading, category);
    return (
        <div className='grid'>
            <h3 className='animate__animated animate__fadeInBottomRight'>{category}</h3>
            {loading && <p className='animate__animated animate__flash animate__delay-1s'>Cargando...</p>}
            <div className='category'>{getCardGifs()}</div>
        </div>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


