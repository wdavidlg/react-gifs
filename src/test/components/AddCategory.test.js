import {AddCategory} from '../../components/AddCategory'
import React from 'react';
import {shallow} from 'enzyme'


describe('pruebas con AddCategory', () => {

    // Se guarda la referencia del metodo setCategories
    const setCategories = jest.fn();

    let wrapper = shallow(<AddCategory 
        setCategories={setCategories}/>);

    beforeEach( () => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory 
        setCategories={setCategories}/>);
    })

    test('debe mostrar el componente correctamente', () => {
        expect( wrapper).toMatchSnapshot();
    })

    test('debe cambiar el valor del input', () => {
        wrapper.find('input').simulate('change', {target: {value: 'mundo'}});
        const valor = wrapper.find('input').prop('value');
        console.log(valor);
        expect(valor).toBe('mundo');
    })
    
    
   test('No debe postear la informacion con submit', () => {
       // preventDefault: () =>{}
       wrapper.find('form').simulate('submit', { preventDefault(){}});
       expect( setCategories ).not.toHaveBeenCalled();

   })
   
   test('deben de llamar el setCategories y limpiar la caja de texto', () => {
       const value = 'hola';
       //Simular el change en el input
       wrapper.find('input').simulate('change', {target:{value}})
       //Simular el submit
       wrapper.find('form').simulate('submit', {preventDefault(){}})
       //setCategories se debe haber llamado
       //similar: toHaveBeenCalledTimes(1)
       expect( setCategories ).toHaveBeenCalled();
       //el value del input debe estar vac√≠o
       expect( wrapper.find('input').prop('value')).toBe('') 
       
   })
   
    
    
})

