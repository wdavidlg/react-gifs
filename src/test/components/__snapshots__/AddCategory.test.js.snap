// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con AddCategory debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
